tests/sles4sap/publiccloud/qesap_ansible.pm

```diff
@@ -22,6 +22,7 @@ sub test_flags {
 
 sub run {
     my ($self, $run_args) = @_;
+    my $provider = $self->provider_factory();
```

Comment:
I think this declaration can go after L39 (perhaps, just before L67) and that way we avoid creating a structured which is not needed when the module runs without failures.

Reply:
Good suggestion. Done.